src/brogue/Buttons.c

```diff
@@ -109,12 +109,11 @@ void drawButton(brogueButton *button, enum buttonDrawStates highlight, cellDispl
             symbolNumber++;
         }
 
-        if (coordinatesAreInWindow(button->x + i, button->y)) {
+        if (locIsInWindow((windowpos){ button->x + i, button->y })) {
+            plotCharToBuffer(displayCharacter, (windowpos){ button->x + i, button->y }, &fColor, &bColor, dbuf);
             if (dbuf) {
-                plotCharToBuffer(displayCharacter, button->x + i, button->y, &fColor, &bColor, dbuf);
+                // Only buffers can have opacity set.
                 dbuf[button->x + i][button->y].opacity = opacity;
-            } else {
-                plotCharWithColor(displayCharacter, button->x + i, button->y, &fColor, &bColor);
```

Comment:
Looking at these lines, as requested - seems fine, if `dbuf` is NULL then plotCharToBuffer calls plotCharWithColor